src/main/java/org/opensearch/knn/index/memory/NativeMemoryLoadStrategy.java

```diff
@@ -119,6 +123,55 @@ public NativeMemoryAllocation.IndexAllocation load(NativeMemoryEntryContext.Inde
             );
         }
 
+        @Override
+        public NativeMemoryAllocation.IndexAllocation load(NativeMemoryEntryContext.IndexEntryContext indexEntryContext)
+            throws IOException {
+            final Path absoluteIndexPath = Paths.get(indexEntryContext.getKey());
+            final KNNEngine knnEngine = KNNEngine.getEngineNameFromPath(absoluteIndexPath.toString());
+            if (knnEngine != KNNEngine.FAISS) {
+                // We will support other non-FAISS native engines (ex: NMSLIB) soon.
+                return loadWithAbsoluteIndexPath(indexEntryContext);
+            }
+
+            final FileWatcher fileWatcher = new FileWatcher(absoluteIndexPath);
+            fileWatcher.addListener(indexFileOnDeleteListener);
+            fileWatcher.init();
+
+            final Directory directory = indexEntryContext.getDirectory();
+
+            // Ex: Input -> /a/b/c/_0_NativeEngines990KnnVectorsFormat_0.vec
+            // Output -> _0_NativeEngines990KnnVectorsFormat_0.vec
+            final String logicalIndexPath = absoluteIndexPath.getFileName().toString();
+
+            final int indexSize = (int) directory.fileLength(logicalIndexPath);
```

Comment:
should this be int or long?

Reply:
On a second thought, it should be long then casted into `int` after divided 1024.
Ex:
```
                return new NativeMemoryAllocation.IndexAllocation(
                    executor,
                    indexAddress,
                    (int)(indexSize / 1024),  // Convert bytes in KB unit. <-------------- This
                    knnEngine,
                    absoluteIndexPath.toString(),
                    indexEntryContext.getOpenSearchIndexName(),
                    watcherHandle,
                    sharedIndexState,
                    IndexUtil.isBinaryIndex(knnEngine, indexEntryContext.getParameters())
                );
```